lib/email_processor.rb

```diff
@@ -6,6 +6,13 @@ def initialize(email)
   end
 
   def process
+
+    # Guard clause to prevent ESPs like Sendgrid from posting over and over again
+    # if the email presented is invalid and generates a 500.  Returns a 200
+    # error as discussed on https://sendgrid.com/docs/API_Reference/Webhooks/parse.html
+    # This error happened with invalid email addresses from PureChat
+    return if @email.from[:email].match(/\A[^@\s]+@([^@\s]+\.)+[^@\s]+\z/).blank?
```

Comment:
There is a way to put all this logic inside a method in `@email` class ? And then here call something like `return unless @email.valid?`